github/github.go

```diff
@@ -427,10 +443,18 @@ func CheckResponse(r *http.Response) error {
 	if err == nil && data != nil {
 		json.Unmarshal(data, errorResponse)
 	}
-	if r.StatusCode == http.StatusUnauthorized && strings.HasPrefix(r.Header.Get(headerOTP), "required") {
+	switch {
+	case r.StatusCode == http.StatusUnauthorized && strings.HasPrefix(r.Header.Get(headerOTP), "required"):
 		return (*TwoFactorAuthError)(errorResponse)
+	case r.StatusCode == http.StatusForbidden && r.Header.Get(headerRateRemaining) == "0" && strings.HasPrefix(errorResponse.Message, "API rate limit exceeded for "):
+		return &RateLimitError{
+			Rate:     parseRate(r),
```

Comment:
I'll keep it as is, because `CheckResponse` is exported and can't easily be modified, and I don't think it's worth adding a second version just to optimize this really minor calculation (at the expense of adding code complexity). Rate limit errors should not occur in "hot loops" where each millisecond saved is beneficial.